src/EditTableDialog.cpp

```diff
@@ -144,8 +144,15 @@ EditTableDialog::EditTableDialog(DBBrowserDB& db, const sqlb::ObjectIdentifier&
                 {
                     // Remove the constraint with the old columns and add a new one with the new columns
                     m_table.removeConstraint(constraint);
-                    constraint->setColumnList(new_columns);
-                    m_table.addConstraint(constraint);
+                    if(constraint->type() == sqlb::Constraint::ForeignKeyConstraintType)
+                    {
+                        auto fk = std::dynamic_pointer_cast<sqlb::ForeignKeyClause>(constraint);
+                        fk->setColumnList(new_columns);
+                        m_table.addConstraint(fk);
+                    } else {
+                        constraint->setColumnList(new_columns);
+                        m_table.addConstraint(constraint);
+                    }
```

Comment:
No need for this change. In both branches of the ```if``` statement you are effectively doing the same thing. So you can just stick with the more general version from before.